covid_graphs/covid_graphs/country_graph.py

```diff
@@ -24,17 +26,35 @@ def __str__(self):
         return self.value
 
 
+@dataclass
+class Trace:
+    xs: List[datetime.date]
+    ys: np.ndarray
+    x_max: datetime.date
+    y_max: float
+    label: str
+
+
+def _create_trace(trace_generator: TraceGenerator, display_until: datetime.date) -> Trace:
```

Comment:
This function and `TraceGenerator.generate_trace` can be merged:
* `Trace` moves to formula.py
* `TraceGenerator.generate_trace` returns a `Trace`

That seems to me more logical.

Reply:
Done